File: 0206-reverse-linked-list/0206-reverse-linked-list.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var pointer :ListNode? = head
        var prev : ListNode? = nil
        var cur : ListNode? = pointer
        while(cur != nil){
            pointer = cur
            cur = cur!.next
            pointer!.next = prev
            prev = pointer
        }
        return pointer
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None:
            return None
        
        pointer = head
        cur = head
        prev = None
        while (pointer):
            pointer = pointer.next
            cur.next = prev
            prev = cur
            cur = pointer
        return prev